### ISSUE TITLE
test get-currentexe fails on AIX

### ISSUE THREAD

**Issue by richardlau** [libuv/libuv#463]
On v1.x the get-current-exe test fails on AIX:

```
-bash-4.2$ ./out/Debug/run-tests
...
[%  58|+ 154|-   5|T   0|S   0]: get_currentexe
`get_currentexe` failed: exit code 393350
Output from process `get_currentexe`:
Assertion failed: __EX, file  ../src/unix/aix.c, line 340
=============================================================
```

It looks like it's failing in the [size=1](https://github.com/libuv/libuv/blob/v1.6.1/test/test-get-currentexe.c#L70) part of the test that was added by 885b1ecda0037fcb6647e8d69a914347ad884244. 

I'm actually running these tests prior to submitting a pull request to address an issue with uv_execpath and relative paths on AIX. If I comment out the  [size=1](https://github.com/libuv/libuv/blob/v1.6.1/test/test-get-currentexe.c#L70) and [size=2](https://github.com/libuv/libuv/blob/v1.6.1/test/test-get-currentexe.c#L78) cases then the test passes:
```
-bash-4.2$ ./out/Debug/run-tests get_currentexe get_currentexe
-bash-4.2$
```

but not if I run the test like this (this is the problem I was originally intending to fix):
```
-bash-4.2$ ../libuv/out/Debug/run-tests get_currentexe get_currentexe
Assertion failed in ../test/test-get-currentexe.c on line 53: match && !strcmp(match, path)
IOT/Abort trap (core dumped)
-bash-4.2$
```

(n.b. I had to invoke with get_currentexe twice on the command line to induce the failure -- I believe the tests are spawned slightly differently in the no argument/single argument cases). 

I'll look to address these failures in my pull request.

**Comment by richardlau** [comment]
For completeness, without any modifications:

Absolute path to run-tests (assertion fails at same line as running run-tests in no argument/single argument case):
```
-bash-4.2$ /home/users/riclau/sandbox/github/libuv/out/Debug/run-tests get_currentexe get_currentexe
Assertion failed: __EX, file  ../src/unix/aix.c, line 340
IOT/Abort trap (core dumped)
-bash-4.2$
```

Relative path to run-tests:
```
-bash-4.2$ out/Debug/run-tests get_currentexe get_currentexe
Assertion failed: __EX, file  ../src/unix/aix.c, line 394
IOT/Abort trap (core dumped)
-bash-4.2$
```

Relative path to run-tests starting with single "./":
```
-bash-4.2$ ./out/Debug/run-tests get_currentexe get_currentexe
Assertion failed: __EX, file  ../src/unix/aix.c, line 371
IOT/Abort trap (core dumped)
-bash-4.2$
```

Relative path to run-tests with "../":
```
-bash-4.2$ ../libuv/out/Debug/run-tests get_currentexe get_currentexe
Assertion failed in ../test/test-get-currentexe.c on line 53: match && !strcmp(match, path)
IOT/Abort trap (core dumped)
-bash-4.2$
```

As before I'll look to address all of these.

**Comment by richardlau** [comment]
Fixed by #464.
